src/main/java/org/terasology/hunger/HungerAuthoritySystem.java

```diff
@@ -82,26 +84,40 @@
     @In
     private Time time;
 
+    /**
+     * Reference to DelayManager, used for addPeriodicAction.
+     */
+    @In
+    private DelayManager delayManager;
+    /**
+     * The interval (in milliseconds) at which healthDecreaseAmount (above) is applied to the component.
+     */
+    public int healthDecreaseInterval = 30000;
+
+    public static final String HUNGER_DAMAGE_ACTION_ID = "Hunger Damage";
     private boolean destroyDrink = false;
 
+    public void postBegin(){
+        for(EntityRef entity : entityManager.getEntitiesWith(WorldComponent.class)){
+            delayManager.addPeriodicAction(entity, HUNGER_DAMAGE_ACTION_ID, 0, healthDecreaseInterval);
```

Comment:
If, on the off chance, there were multiple entities with the `WorldComponent` this would run multiple times.
This would then cause the function to be run for each of these registered delays. I think you only want _one_ entity rather than all. I'm not sure of the best way to do that in this case.

Additionally it may be worthwhile having multiple registered for if/when we get multiworld. However I feel that would be something best ot approach then.

Reply:
I had the view that only one entity with WorldComponent will be present, and multiple such entities would be present in multiworld setup.
Also, `entityManager.getEntitiesWith()` returns iterable of entities with required component (I don't think breaking after adding periodicAction in first such entity would be a good implementation, and i don't know how it will work out in multiworld setup)
Any ideas on how to improve this?

Reply:
Yeah I made the assumption that there'll only be one world entity _for now_ and perhaps even after multi-world is fully added as my expectation there is that each world will have its own entire entity _pool_ so again each world entity will be on its own.

Maybe to guard against mysteries only process the first entity found and if for some odd reason more are found log some details and skip processing?